src/main/java/ch/uzh/ifi/hase/soprafs22/entity/BlockedUserRelation.java

```diff
@@ -0,0 +1,76 @@
+package ch.uzh.ifi.hase.soprafs22.entity;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.server.ResponseStatusException;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+import java.io.Serializable;
+import java.util.*;
+
+/**
+ * This Entity Class is used as a relationship class that holds the relationship if a user blocks another user. Basically,
+ * it works exactly like Matches, therefore, if a user deletes his / her profile, the relationship gets deleted on
+ * cascade. It creates two tables: One with the id and the creationDate and one with the relationship between the users.
+ */
+@Entity
+@Table(name = "BLOCK_RELATION")
+public class BlockedUserRelation implements Serializable {
+  @Id
+  @GeneratedValue
+  @Column(name = "id")
+  private long blockedRelationId;
+
+  // In order that the block_relation table does not only hold the id ...
+  @Column(name = "creation_date")
+  private final Date creationDate = new Date();
+
+  @ManyToMany(fetch = FetchType.LAZY,
+          cascade = {CascadeType.DETACH, CascadeType.MERGE, CascadeType.PERSIST, CascadeType.REFRESH})
+  @JoinTable(name = "BLOCKED_USER_RELATION",
+          joinColumns = {@JoinColumn(referencedColumnName = "id")},
+          inverseJoinColumns = {@JoinColumn(referencedColumnName = "user_id")})
+  private Set<User> users = new HashSet<>();
```

Comment:
instead of a help entity blockedUserRealtion, would it also be possibel for user to have a manytomany relation with itself?
only interest (same for matches)

Reply:
For blockedUsers, that was what we had before. Since you have to update both Users, this would lead to errors à gogo. You would have a table like User1_id | User2_id, with two tuples that are just the inverse of each other, which would (and has) lead to constraintviolation error nevertheless. This is also the reason why I created the Relation Entity, because now there is a bidirectional relationship between User and BlockedUserRelationship, where the later IS the owner of User (which might at first seem counterintuitiv, but makes perfect sense if you think about the tables). 
For the matches, I don't see why this would make sense, since a match also holds other important things like the chatId.

Reply:
yes match was a class in our class diagram. as i tryed to break the responsibilities for the user class a little.
just was interested:)